### PR TITLE
Prepare the package for the new Surface notation

### PR DESCRIPTION
**Pull request by Malian**
- [x] Set the Surface version to 0.5.0 as soon as it is release
- [x] Update the README to point to the correct version depending on the Surface version the user uses

### CODE REVIEW

File: lib/heroicons/components/icon.ex [hunk 1]
```diff
@@ -19,7 +19,7 @@ if Code.ensure_loaded?(Surface) do
         end
 
       ~H"""
```
Comment (miguel-s) [inline lib/heroicons/components/icon.ex]:
```suggestion
      ~F"""
```

File: mix.exs [hunk 2]
```diff
@@ -24,7 +24,8 @@ defmodule Heroicons.MixProject do
     [
       {:phoenix_html, "~> 2.14"},
       {:ex_doc, "~> 0.24", only: :dev, runtime: false},
-      {:surface, "~> 0.4", optional: true}
+      {:surface,
+       git: "https://github.com/surface-ui/surface", branch: "surface-next", optional: true}
```
Comment (miguel-s) [inline mix.exs]:
```suggestion
      {:surface, "~> 0.5", optional: true}
```

**Comment by miguel-s** [comment]
@Malian I know that you're very busy these days so i took the liberty to add some changes to your PR. I think it's ready to merge WDYT?

**Comment by Malian** [comment]
@miguel-s Feel free to go forward without me, after all, it's your open source project ;)

The last thing we should do is to update the README to point to the `v0.3.0-dev` of the package. Not sure we should do this on the PR or when you think the `v0.3.0` is ready.

Except this point, LGTM!

**Pull request merged**